lib/apis/moray.js

```diff
@@ -492,7 +492,33 @@ Moray.prototype._createSearch = function (params, cb) {
     });
 };
 
+/**
+ * List all VMs for a given server from the VM bucket.
+ * @param  {string}     uuid uuid of the server whos vms should be fetched.
+ * @param  {Function}   cb  function of the form function(err, vms) {...}
+ */
+Moray.prototype.listVmsForServer = function listVmsForServer(uuid, cb) {
+    var self = this;
+    var vms = {};
+    var vm;
+    var filter = sprintf(PARAM_FILTER, 'server_uuid', uuid);
```

Comment:
I think you'll want to filter here on the active states. Even with active VMs you need to assume that the results here could be up to ~2000, but without the state filter this will include all VMs that have ever been created on the CN (all the destroyed + active). Which could be a pretty big number, especially for Triton/Docker CNs.